### ISSUE TITLE
Feature branch deploys access denied, but main branch is fine

### ISSUE THREAD

**Issue by chris** [aws-amplify/amplify-console#740]
I have a Nuxt/Vue app where I have feature branch auto-deploys configured. My main branch deploys just fine, site works, etc. However, the feature branches all result in an AccessDenied error. This may be related to #436 or #75 or #526 or #591, however I tried adding the SPA redirect rule and that did not fix it. Further, it seems (maybe) this is specific to feature branches since the main branch deploy works fine. And also, I'm doing Nuxt generate, so it should be a fully static site, with no routing aspects, etc.

Here is may amplify.yml:
```
version: 0.1
frontend:
  phases:
    preBuild:
      commands:
        - yarn install
    build:
      commands:
        - yarn run generate
  artifacts:
    baseDirectory: dist
    files:
      - '**/*'
  cache:
    paths:
      - node_modules/**/*
```

As an example of the error page I get, it is:
```
This XML file does not appear to have any style information associated with it. The document tree is shown below.
<Error>
<Code>AccessDenied</Code>
<Message>Access Denied</Message>
<RequestId>87410EAC7CEC7C13</RequestId>
<HostId>
JK1Elswet0Laks3CW0U8aZWZa78DMMWLpTfY9CuLL4aNTqpheyJ+uW7Xz5uCG2rGPWAE6lKb8WQ=
</HostId>
</Error>
```

**Comment by Joycehao19** [comment]
HI @chris  would you mind share your appId, branch name and region with us?

**Comment by chris** [comment]
@Joycehao19 the app ID is `d1zq9cj8sfb2r4`, and region is us-east-2. There are currently two branches (which are all the branches) that experience this: "feature/amplify-build-config" and "feature/skeletal-html".

**Comment by swaminator** [comment]
@chris did you figure it out?

**Comment by chris** [comment]
@swaminator no, waiting on @Joycehao19 to see if they can clarify what the problem is.

**Comment by swaminator** [comment]
@chris can you try this out: https://github.com/aws-amplify/amplify-console/issues/436#issuecomment-629872652

**Comment by chris** [comment]
@swaminator oddly, we haven't changed anything, but now our branches are rendering properly. Even more oddly - the branch that I had that wasn't working right, with no changes, is now rendering! So, I'm at a bit of a loss as to what was going on and/or what fixed it.

**Comment by swaminator** [comment]
@chris we recently released a fix for dot files https://github.com/aws-amplify/amplify-console/issues/278 don't know if that somehow addressed this issue as well? Closing this for now. Please open a new issue if you encounter this again.

**Comment by Liguo-Ma** [comment]
Hello 
I have the same error too.
The main domain works well.
But if I redirect the subdomain I get AccessDenied error.
For example, prolocalize.com works well but prolocalize.com/login does not work

**Comment by chris** [comment]
@Liguo-Ma I would try doing a redeploy. I filed an AWS support ticket and they said Amplify had some bug they fixed and to try deploying again now. I did so, and this problem went away for me. It's very disappointing that there seems to be frequent bugs with URLs/routing/branch deploys in Amplify (this is definitely not the first or second time I've had such problems), but there is no status update or anything from Amplify indicating there's currently a problem.

**Comment by Liguo-Ma** [comment]
Thanks @chris  
But now main domain do not work too :(
Could you let me know what is the solution?
And please let me know how to send ticket to AWS support about issue?

**Comment by chris** [comment]
@Liguo-Ma Support tickets can be filed via your regular AWS account (assuming you're paying for a support plan).

If your main domain is not working then I'm not sure, that sounds like a different bug, or maybe an issue with your app? Have you looked at the various logs for your Amplify builds/deploys? Are they all successful (there are multiple different logs, e.g. build, deploy, etc.)?

**Comment by Liguo-Ma** [comment]
Hello @chris 
I have tried to publish again but  I get first error.
So Main domain works fine but after redirect subdomain I get  <Code>AccessDenied</Code> error
Please let me know what is the solution.

**Comment by chris** [comment]
@Liguo-Ma I'm not sure what that would be. I would file a support case with AWS, since it seems it is something different. Sorry this is happening. I realize I filed this issue here back in June, yet this happened again recently (just last week, which is why I filed the AWS support ticket). Seems to be a recurring problem for Amplify. Since this GitHub issue is closed, I doubt anyone from AWS is looking at it, so again, I'd file an AWS account support ticket (or try opening a new issue here on GitHub and reference this issue). For the AWS support ticket, you can reference my ticket: 7386358291. I don't know if they're the same issue (since mine is now fixed, seems like it may be something different, but maybe related?).

**Comment by Liguo-Ma** [comment]
Yes, you are right.
It worked fine last month.
So it is the issue of the AWS team?
Also, Could you let me know soon once they will fix?

**Comment by Liguo-Ma** [comment]
Oh, I am using the old version Amplify CLI.
Is it the reason for this error?

**Comment by chris** [comment]
@Liguo-Ma I am just an AWS customer, I'm not on the AWS team nor have insight into them. I'm not using the Amplify CLI (my builds are triggered via GitHub pushes), but I doubt that is part of it. Again, I'd file an AWS support ticket via the AWS console.

**Comment by Liguo-Ma** [comment]
Okay  I see
I have tried with the lastest amplify version but it does not work too.
For example, https://prod.d3gsscuikbv8vv.amplifyapp.com/ navigate to https://prod.d3gsscuikbv8vv.amplifyapp.com/signin/.
But redirect https://prod.d3gsscuikbv8vv.amplifyapp.com/signin/ get error.
I really sure what is the solution.
Could you let me know why this happened?

**Comment by chris** [comment]
Is this a change in behavior, or is your signin page a new addition to your app? Since it navigates ok from the root to the signin page, this feels to me like the problem where *directly* navigating to a page doesn't work because there is no actual page file for that page (e.g. there isn't a signin.html that is generated for your app). That, or you have rewrite rules that are causing problems. I haven't used Ionic, so don't know how that works. But, presuming it is generating a static site from your code, this feels like the case where there is no corresponding page file, so Amplify can't navigate to it. If there is no signin.html file, then S3/Cloudfront has no page to serve. It works when navigating from your root URL in these cases because the JavaScript app has been loaded and its own router is used to "load" that "page".

**Comment by Liguo-Ma** [comment]
I do not think so.
I uploaded the folder after building it manually.
And https://prod.d3gsscuikbv8vv.amplifyapp.com/  navigate https://prod.d3gsscuikbv8vv.amplifyapp.com/ signin/ because there is Signing page .
But If I redirect signing page get error.
It is very strange.
I did not see this error 1 month ago :(
